homeassistant/components/lastfm/coordinator.py

```diff
@@ -0,0 +1,65 @@
+"""Data update coordinator for the LastFM integration."""
+from dataclasses import dataclass
+
+from pylast import SIZE_SMALL, LastFMNetwork, Track, WSError
+
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_USERS, DOMAIN, LOGGER, STATE_NOT_SCROBBLING
+
+
+def format_track(track: Track) -> str:
+    """Format the track."""
+    return f"{track.artist} - {track.title}"
+
+
+@dataclass
+class UserData:
+    """Data object for transfer to the sensors."""
+
+    image: str
+    now_playing: str
+    top_played: str | None
+    last_played: str
+    play_count: int
+
+
+class LastFmUpdateCoordinator(DataUpdateCoordinator[dict[str, UserData]]):
+    """Data update coordinator for the LastFM integration."""
+
+    def __init__(self, hass: HomeAssistant, config: ConfigType) -> None:
+        """Initialize the coordinator."""
+        super().__init__(hass=hass, logger=LOGGER, name=DOMAIN)
+        self._users = config[CONF_USERS]
+        self.lastfm_api = LastFMNetwork(api_key=config[CONF_API_KEY])
+
+    def _update(self) -> dict[str, UserData]:
+        response = {}
+        for user in self._users:
+            try:
```

Comment:
I'm starting to think the implementation is not going the right way.

Does it really make sense to make one massive update for all users in the same call?

If in the future you deactivate an entity you would still be querying the user, image, now_playing etc for inactive entities as well as active entities.

You are also loosing here the "failure" aspect. If one user fails to download, then there is nothing to mark the entity as unavailable. There is only an error in the log, with the entity keeping the old state.

Reply:
Yep, I really liked my first PR since that was a drop in replacement including integration into the old config and allowing the new config to also be there. Now the coordinator is hard coupled to the old config which should be changed later and stuff. 

What direction should we go with this? Do you have an example integration to look at? All these changed are mostly based on the steam_online integration.

Reply:
#91601 has the same problems. Having a data coordinator doesn't make sense there either.
I suggest that you close this PR, and remove the update coordinator altogether from #91601

Reply:
Or you make the coordinator user-specific (create a separate coordinator instance for each entity but I think that may be overkill).

Reply:
Do you have an integration that looks like the idea you have in your head? I am always finding trouble when I work with data since every integration does it (slightly) differently, so I am never sure what is the best way forward